File: BackEnd/SpringBootServer/src/main/java/com/ssafy/backend/domain/news/entity/News.java
package com.ssafy.backend.domain.news.entity;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import com.ssafy.backend.domain.news.entity.enums.NewsCategory;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Builder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PROTECTED)
public class News {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(columnDefinition = "INT UNSIGNED")
    private Long id;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String titleTranslated;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private NewsCategory category;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String contentTranslated;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String summary;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String summaryTranslated;

    @Column(updatable = false, nullable = false, columnDefinition = "TIMESTAMP")
    private LocalDateTime writeDate;

    @Column(nullable = false)
    private String srcOrigin;

    @OneToMany(mappedBy = "news", fetch = FetchType.LAZY)
    private List<NewsImage> newsImages = new ArrayList<>();

    @OneToMany(mappedBy = "news", fetch = FetchType.LAZY)
    private List<NewsKeywordMapping> newsKeywordMappings = new ArrayList<>();

    @OneToMany(mappedBy = "news", fetch = FetchType.LAZY)
    private List<NewsShadowing> newsShadowing = new ArrayList<>();
}

File: BackEnd/FastApiServer/models/news.py
# models/news.py
from sqlalchemy import Column, BigInteger, String, Text, DateTime
from sqlalchemy.orm import relationship
from database import Base

class News(Base):
    __tablename__ = "news"

    id = Column(BigInteger, primary_key=True, index=True)
    title = Column(String(255), nullable=False)
    category = Column(String(255), nullable=False)
    content = Column(Text, nullable=False)
    summary = Column(Text, nullable=False)
    write_date = Column(DateTime, nullable=False)
    news_images = relationship("NewsImage", back_populates="news")
    news_keyword_mappings = relationship("NewsKeywordMapping", back_populates="news")